src/app.rs

```diff
@@ -206,82 +207,87 @@ impl Application {
+                            addr: api_config.address.unwrap(),
+                            playground: api_config.playground
+                        });
+
+                        Some(api::Server::start(topology.config(), topology.watch()))
+                    } else {
+                        info!(message="API is disabled, enable by setting `api.enabled` to `true` and use commands like `vector top`.");
+                        None
+                    };
+                }
+            );
 
             let signals = signal::signals();
             tokio::pin!(signals);
             let mut sources_finished = topology.sources_finished();
 
             let signal = loop {
                 tokio::select! {
-                Some(signal) = signals.next() => {
-                    if signal == SignalTo::Reload {
-                        // Reload paths
-                        config_paths = config::process_paths(&opts.config_paths_with_formats()).unwrap_or(config_paths);
-                        // Reload config
-                        let new_config = config::load_from_paths(&config_paths).map_err(handle_config_errors).ok();
-
-                        if let Some(mut new_config) = new_config {
-                            new_config.healthchecks.set_require_healthy(opts.require_healthy);
-                            match topology
-                                .reload_config_and_respawn(new_config)
-                                .await
-                            {
-                                Ok(true) => {
-                                    #[cfg(feature="api")]
-                                    if let Some(ref api_server) = api_server {
-                                        api_server.update_config(topology.config())
+                    Some(signal) = signals.next() => {
+                        if signal == SignalTo::Reload {
+                            // Reload paths
+                            config_paths = config::process_paths(&opts.config_paths_with_formats()).unwrap_or(config_paths);
+                            // Reload config
+                            let new_config = config::load_from_paths(&config_paths).map_err(handle_config_errors).ok();
+
+                            if let Some(mut new_config) = new_config {
+                                new_config.healthchecks.set_require_healthy(opts.require_healthy);
+                                match topology
+                                    .reload_config_and_respawn(new_config)
+                                    .await
+                                {
+                                    Ok(true) => {
+                                        #[cfg(feature = "api")]
+                                        // Pass the new config to the API server.
+                                        if let Some(ref api_server) = api_server {
+                                            api_server.update_config(topology.config());
```

Comment:
Ideally, this would updated to follow the same pattern. It's a fair amount of work, since metrics subscriptions are built around the notion of global state. I think this is a good follow-up issue. Tracking in #6939.